packages/@glimmer/runtime/lib/compiled/opcodes/dom.ts

```diff
@@ -194,15 +144,114 @@ function invokeModifier(
 
   if (tag !== null) {
     consumeTag(tag);
-    vm.updateWith(new UpdateModifierOpcode(tag, instance));
+    return vm.updateWith(new UpdateModifierOpcode(tag, instance));
   }
-}
+});
+
+APPEND_OPCODES.add(Op.DynamicModifier, (vm) => {
+  if (vm.env.isInteractive === false) {
+    return;
+  }
+
+  let { stack, [CONSTANTS]: constants } = vm;
+  let ref = check(stack.popJs(), CheckReference);
+  let args = check(stack.popJs(), CheckArguments).capture();
+  let { constructing } = vm.elements();
+  let initialOwner = vm.getOwner();
+
+  let instanceRef = createComputeRef(() => {
+    let value = valueForRef(ref);
+    let owner: Owner;
+
+    if (typeof value !== 'function' && (typeof value !== 'object' || value === null)) {
+      return;
+    }
+
+    let hostDefinition: CurriedValue | ModifierDefinitionState;
+
+    if (isCurriedType(value, CurriedType.Modifier)) {
+      let {
+        definition: resolvedDefinition,
+        owner: curriedOwner,
+        positional,
+        named,
+      } = resolveCurriedValue(value);
+
+      hostDefinition = resolvedDefinition;
+      owner = curriedOwner;
+
+      if (positional !== undefined) {
+        args.positional = positional.concat(args.positional) as CapturedPositionalArguments;
+      }
+
+      if (named !== undefined) {
+        args.named = assign(named, args.named);
```

Comment:
Chatted with @pzuraq RE: this, and `named` is a fresh object created by `resolveCurriedValue` which is why this mutation is safe.

Reply:
One thing that struck me, is that we could pass `args.named` in to `resolveCurriedValue` and avoid multiple `Object.assign`s.